doc/architecture/architecture.rst

```diff
@@ -1,24 +1,234 @@
+
+#. **Ostro OS is easy to customize**: While Ostro OS comes pre-compiled
+   (binaries and images), it can be customized
+   to meet the diverse product needs of IoT applications. With the help of
+   the Yocto Project build system used by Ostro OS, it is easy to customize
+   the build and generated images that are finely tuned to meet your exact needs.
+   :ref:`about_ostro` and the links provided there are an excellent starting point
+   to learn more about the Ostro OS extensive customization possibilities. These
+   pre-compiled images are intended to expedite your development work and not intended
+   for use in production systems.
+
+Ostro OS Architecture Stack
+===========================
+
+One way to describe Ostro OS architecture is to build a stack from the
+hardware platform up to an IoT Application layer. This stack is
+illustrated here.
+
+.. image:: images/ostro_os_architecture.svg
+   :width: 400px
+
+The following sections examine each layer in more details.
+
+IoT Applications
+----------------
+The IoT applications layer consist of all applications that use the
+underlying platform. Ostro OS itself currently does not
+implement any specific IoT use-cases via applications, but 
+`application sample recipes`_ are provided to demonstrate the OS frameworks and services.
+
+Programming Interfaces
+-----------------------
+
+Ostro OS provides various application runtimes offering more flexibility and
+options to application developers to implement their IoT solution.
+
+* `Soletta Project`_ is an open-source
+  framework for making IoT applications. The project provides libraries
+  to make it quick and easy to write software for IoT devices. The
+  applications can we written using flow-based programming (FBP) or
+  more traditionally as C applications linking to Soletta's C-based
+  platform APIs.
+
+* `Node.JS`_ is a popular JavaScript runtime to
+  run (IoT) web applications. Ostro OS provides the Node.JS runtime
+  and selected JavaScript APIs (as Node.JS modules) to build IoT
+  applications. The set of APIs includes, for example, a Javascript API for
+  the `Open Connectivity Foundation`_ (OCF) specifications. The JavaScript
+  API set is kept aligned with the API set provided by Soletta.
+
+* Ostro OS also provides support for the popular IoT-space development
+  languages Python and Java\* (OpenJDK8).
+
+Frameworks
+----------
+
+Frameworks provide an abstraction to platform services, making
+application deployment easier. The Ostro Application
+Framework :ref:`application-framework` provides tooling for
+application writers to get the applications running in an Ostro OS
+based image.
+
+The Ostro Application framework implements a `systemd.generator'_ that
+parses the application manifest files to generate a systemd service files.
+
+Ostro OS also pre-integrates `IoTivity`_ framework. IoTivity is an open source
+project that implements the full OCF specification by _`Open Connectivity Foundation`. In
```

Comment:
move the underscore after the `name` for an outgoing link:  should be `Open Connectivity Foundation`_